jgrapht-core/src/main/java/org/jgrapht/alg/cycle/DirectedSimpleCycles.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (C) Copyright 2013-2017, by Nikolay Ognyanov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+/**
+ * A common interface for classes implementing algorithms for enumeration of the simple cycles of a
+ * directed graph.
+ *
+ * @param <V> the vertex type.
+ * @param <E> the edge type.
+ *
+ * @author Nikolay Ognyanov
+ */
+@Deprecated
+public interface DirectedSimpleCycles<V, E>
```

Comment:
JGraphT uses a 2-release deprecation: 
  - an interfaces becomes deprecated (retaining all methods)
  - next release
  - the interface is deleted
  - next release
  
Also needs `@deprecated` comment in Javadoc with link to the next interface. See other code in the library for examples.